docs/proposal-spawn-on-d.md

```diff
@@ -205,22 +205,51 @@ trigger events too, but for now a waiting task with an xtrigger is fine).
 
 ### Absolute dependence
 
-Here `foo.1,2,3,...` should trigger off of `start.2`:
+At first glance this is not a great fit for the spawn-on-demand model: 
 ```
 ...
       R1/2 = "start"
-      P1 = "start[2] => foo"
+      P1 = "start[2] => bar"
 ```
+Above, with no final cycle point a single event (`start.2 succeeded`) should
+spawn an infinite number of children (`bar.1,2,3,...`), which obviously is not
+feasible.
+
 (Note without the first line above `start` is not defined on any sequence, so
-with SoS the scheduler will stall with `foo` unsatisfied, but with SoD it will
-shut down immediately with nothing to do because and `foo` never gets spawned.)
+with SoS the scheduler will stall with `bar` unsatisfied, but with SoD it will
+shut down immediately with nothing to do because and `bar` never gets spawned.)
 
-So: when an absolute parent finishes *spawn its children at the first cycle
-point of the sequence.* Above, `start.2` should spawn `foo.1`, not `foo.2`.
-Then whenever an absolute child is released from the runahead pool, spawn
-its next instance (to the runahead pool) and mark the absolute dependence as
-satisfied. This works because even the first child does not get spawned
-until the associated dependence is already satisfied. 
+Worse, `bar` could also have other non-absolute triggers, and it could be
+those that spawn initial instances of `bar` before `start` is finished:
+```
+...
+      R1/2 = "start"
+      P1 = "start[2] & foo => bar"
+```
+Above, if `foo` runs 3x (say) faster than `start` then roughly speaking the
+first three instances of `foo` will spawn the first three instances of `bar`
+before `start.2` is done. This suggests we might need to do repeated checking
+of unsatisfied absolute triggers until they become satisfied. However, it turns
+out that's not necessary...
+
+#### Absolute trigger implementation:
+
+1. Whenever an absolute output (e.g. `start.2:succeed` above) is completed the
+scheduler should:
+   - Remember it forever, via a list held by the task pool module and an
+    associated DB table (to load the data again on restart).
+   - Spawn (if not already spawned) the first downstream child (`bar.1` above)
+    **and** update the prerequisites of potentially multiple subsequent
+    children already in the pool due to spawning by others (e.g. `foo` above).
+2. Whenever a task with an absolute trigger (`bar` above) is spawned, check to
+see if the trigger is already satisfied. If it is, satisfy it. If it isn't then
+it will be satisfied later when the absolute parent finishes and updates all
+child instances present in the pool as described just above. 
+3. Whenever a task with an absolute trigger is released from the runahead
+pool spawn its next instance (strictly speaking this is only necessary for
+tasks with no non-absolute triggers to spawn them on demand).
+
+#### Retriggering
```

Comment:
"retrigger the absolute parent without reflow, then retrigger with reflow the first child that you want to continue from"
That will only run that single child won't it?

Reply:
- "retrigger the absolute parent without reflow" - this will only trigger the absolute parent task
- "retrigger with reflow the first child that you want to continue from" - this will trigger "the first child that you want to continue from" and that will (on release from the runahead pool) spawn the next child, and so on - because that's how we spawn tasks with absolute parents. (Note I've said "spawn" rather than "run" for subsequent child tasks).

Reply:
I'll make the text clearer...